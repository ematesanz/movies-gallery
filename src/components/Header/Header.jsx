import styles from './Header.module.css';

const Input = ({onChange}) => (
  <div className={styles.search__container}>
    <label htmlFor="seach">Search </label>
    <input
      className={styles.search}
      placeholder="What do you fancy today?"
      onChange={onChange}
    />
  </div>
)

const Favorites = ({onClick}) => (
  <div>
    <img className={`${styles.icon} ${styles.favorites}`} alt="Star" onClick={onClick} />
  </div>
)

const WatchLater = ({onClick}) => (
  <div>
    <img className={`${styles.icon} ${styles['watch-later']}`} alt="Watch later" onClick={onClick} />
  </div>
)

const Clear = ({onClick}) => (
    <div>
      <img className={`${styles.icon} ${styles.clear}`} alt="clear" onClick={onClick} />
    </div>
  )

const searchMovie = (value, data, setSearch, search) => {
  switch(search){
    case 'search':
      setSearch(data.filter((movie) => movie.title.toLowerCase().includes(value.toLowerCase())));
    break;
    case 'favorites':
      setSearch(data.filter((movie) => movie.favorite));
    break;
    case 'watchLater':
      setSearch(data.filter((movie) => movie.watchLater));
    break;
    case 'clear':
      setSearch(data);
    break;
    default:
    return;
  }
}


const Header = ({data, setSearch}) => 
  <>
    <div className={styles.header}>
      <Input onChange={event => searchMovie(event.target.value, data, setSearch, 'search')} />
      <Favorites onClick={event => searchMovie(event.target.value, data, setSearch, 'favorites')} />
      <WatchLater onClick={event => searchMovie(event.target.value, data, setSearch, 'watchLater')} />
      <Clear onClick={() => searchMovie('', data, setSearch, 'clear')} />
    </div>
  </>

export default Header;
