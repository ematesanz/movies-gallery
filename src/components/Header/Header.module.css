.header {
  display: flex;
  flex: 1 1;
  padding: 35px 30px;
}

.search__container {
padding-right: 50px;;
}

.search {
  border-left-color: transparent;
  border-right-color: transparent;
  border-top-color: transparent;
  width: 300px;
}

.icon{
  display: inline-block;
  width: 32px;
  height: 32px;
  margin-right: 5px;
  background-size: 100%;
}

img.star {
  content: url("../../img/star-active.png");
}

img.watch-later {
  content: url("../../img/bookmark-active.png");
}