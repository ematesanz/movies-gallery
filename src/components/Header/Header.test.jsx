import {render} from "@testing-library/react";
import Header  from "./Header";

describe('Header component', () => {
  it('renders correctly', () => {
    const {container, getByPlaceholderText} = render(
      <Header/>
    );
    expect(container).toMatchSnapshot();
    expect(getByPlaceholderText('What do you fancy today?')).toBeVisible();
    expect(container.getElementsByClassName('favorites').length).toBe(1);
  });
});
