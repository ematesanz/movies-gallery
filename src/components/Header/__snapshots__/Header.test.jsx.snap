// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="search__container"
    >
      <label
        for="seach"
      >
        Search 
      </label>
      <input
        class="search"
        placeholder="What do you fancy today?"
      />
    </div>
    <div>
      <img
        alt="Star"
        class="icon favorites"
      />
    </div>
    <div>
      <img
        alt="Watch later"
        class="icon watch-later"
      />
    </div>
    <div>
      <img
        alt="clear"
        class="icon clear"
      />
    </div>
  </div>
</div>
`;
